jujugui/static/gui/src/app/app.js

```diff
@@ -3169,15 +3166,11 @@ YUI.add('juju-gui', function(Y) {
         }
         if (auth) {
           this.get('users')[service] = auth;
-          // If the profile is visible then we want to rerender it with the
-          // updated username.
-          if (rerenderProfile) {
```

Comment:
When implementing the new profile page, calling the charmstore whoami to log in was causing race conditions. Instead of implementing a hack to cover it up like we have in the old profile. Removing this and then redispatching the app when the user is stored just makes a lot more sense